File: Arrays/MajorElement/Solution4/main.py
class MajorElement:
    def majElement(self,arr):
        n = len(arr)
        count = 1
        leader = arr[0]
        for i in range(n):
            if count == 0:
                leader = arr[i]
                count = 1
            elif leader == arr[i]:
                count +=1
            else :
                count-=1
        return leader 

n = int(input("Enter the size of arrays: "))
arr = list(map(int,input("Enter the array element: ").split()))
obj = MajorElement()
result = obj.majElement(arr)
print("The majority element is: ",result)

File: Arrays/MajorElement/Solution4/Main.java
package Arrays.MajorElement.Solution4;
import java.util.Scanner;


class MajorElement{
    static int majElement(int[]arr){
        int leader = arr[0];
        int major = 1;
        int n = arr.length;
        for(int i=1;i<n;i++){
            if(arr[i]==leader){
                major++;
            }
            else{
                major--;
            }
        }
       return leader;
    }
    public static void main(String[]args){
        try(Scanner sc = new Scanner(System.in)){
            System.out.print("Enter the size of the array:");
            int n = sc.nextInt();
            System.out.print("Enter the elements of the array:");
            int[] arr = new int[n];
            for(int i=0;i<n;i++){
                arr[i] = sc.nextInt();
            }
            int result = majElement(arr);
            System.out.println("The major element is: " + result);
        }
    }
}